controllers/apps/transformer_component_workload.go

```diff
@@ -641,6 +641,10 @@ func (r *componentWorkloadOps) leaveMember4ScaleIn() error {
 			return switchoverErr
 		}
 
+		if r.synthesizeComp.LifecycleActions == nil || r.synthesizeComp.LifecycleActions.MemberLeave == nil {
```

Comment:
This check should be done before creating lorry clients.

Reply:
there is a switchover before memberleave, if this check is placed before creating lorry client, it could cause a problem when deleting leader node, as cmpd may have a switchover but no memberleave action like mysql replication.

Reply:
For the scale-in operation, calling memberLeave is a functional requirement about the correctness, while switchover is just an optimization about availability. And, the change in this way cannot solve the problems encountered currently.